pkg/ambassador/tap/conduit/conduit.go

```diff
@@ -193,7 +204,10 @@ func (c *Conduit) GetIPs() []string {
 func (c *Conduit) SetVIPs(vips []string) error {
 	c.mu.Lock()
 	defer c.mu.Unlock()
-	if !c.isConnected() {
+	c.ctxMu.Lock()
```

Comment:
Should be a;
```go
if c.ctx != nil { }
```

Reply:
And why the double mutex'es in the first place? `c.mu` is already held, which ensures single thread.